### DISCUSSION TITLE
Line Break in DOCX

### DISCUSSION THREAD

**Discussion by toiepp** [eclipse-birt/birt#1655] [Q&A]
Hi! I am having an issue with BIRT. I'm trying to generate DOCX file based on .rtpdesign file. When I do it as shown below 
```
            case DOCX -> {
                var renderOption = new DocxRenderOption();
                renderOption.setOutputFormat("docx");
                renderOption.setOutputStream(outputStream);
                task.setRenderOption(renderOption);

                log.debug("Set up output format -> docx");
            }

```
BIRT gives me DOCX file, but it ignores line break symbol. What should I do to fix it? Couldn't find any documentation or tutorial.

Thank you!

**Comment by speckyspooky** [comment]
For me is not clear what you mean with "ignores line break symbol".
Do you get the correct DOCX with the designer?

Could you provide us a small example of your report with the topic but without any datasource.

**Comment by toiepp** [comment]
@speckyspooky, Thank you for your response!

By "ignore" I mean that when I have some string (`String str = "Hello,\nworld"`) and I'm trying to put into rtpdesign file using `task.setParameterValue("param", str)`, then in final output file `str` would look like this `Hello,world` (no line break).

This is the only issue I have. In all other things DOCX file looks fine. Not sure if I am allowed to give some real examples, I think what I explained above would be necessary.

**Comment by speckyspooky** [comment]
I haven't exactly your situation,
I tested it only on a standard parameter and one point is that the "line break" will be parsed several times through the engine for the output. The additional thing is that the output is dependend to the element which will be used on report side.

I added your text als parameter-default als scripted text and it was correctly displayed as "dynamic text"
with a standard "data"- or "text (HTML)"-element I got "Hello,<br>world".

Which element do you use on report level?
Could you try "<br>" to use the HTML-line break.

**Comment by hvbtup** [comment]
I'd like to clarify Thomas' point. In a data item, the content is shown as-is. But in a dynamic text item or text item, the content is HTML by default (unless you explicitly define it as plain text), so a line-break character is handled like any other white-space character, so the output should be `Hello, world` (but not `Hello,world`).
